### PR TITLE
[#35] [#36] [UI] [Integrate] As a user, I can see the text field question when taking a survey

### LINKED ISSUE #35: [UI] As a user, I can see the text field question when taking a survey
## Why

App must display the text field question before taking a survey

## Acceptance Criteria

- Must display the text field question
- Number of text fields is based on number of `Answer`s inside `Question`
- Text field hint must be `Question.Answer.Text`
- Text field input type must be based on `Question.Answer.inputMask`
- Keyboard must show "next" button until the last question.
  - On the last question, keyboard must show "done"

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=2632%3A519)

### LINKED ISSUE #36: [Integrate] As a user, I can see the text field question when taking a survey
## Why

App must display the text field question before taking a survey

## Acceptance Criteria

- Must display the text field question
- Must display question with `displayType` "textfield"

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=2632%3A519)

### PR DESCRIPTION
**Pull request by ryan-conway**
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/35
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/36

## What happened 👀

- Created `AnswerTextFieldWidget`
- Updated `BackgroundWidget` with `child` param to fix background resizing issue 👀 

## Insight 📝

If we have `BackgroundWidget` inside `QuestionPagingWidget`, the image will resize when the keyboard is shown. To fix this, we should structure the `SurveyDetailScreen` widget like this: 
```
- Container
  - Image
  - Scaffold
    - QuestionPagingWidget
```

## Proof Of Work 📹

https://github.com/nimblehq/flutter-ic-lydia-ryan/assets/53168251/32500ecb-ed3f-4991-b729-876377a52c41

### CODE REVIEW

File: lib/screens/surveydetails/widget/answer_text_field_widget.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      controller: TextEditingController(),
+    );
+
+    return Center(
+      child: Form(
+        autovalidateMode: AutovalidateMode.always,
+        child: ListView(
+          shrinkWrap: true,
+          children: [
+            ...List<Widget>.generate(
```
Comment (lydiasama) [inline lib/screens/surveydetails/widget/answer_text_field_widget.dart:48]: 😮 👍🏻

**Review by lydiasama — APPROVED** [review]

**Comment by lydiasama** [comment]
@ryan-conway Could you please add `[#35]{#36]` at the PR title too?

File: lib/screens/surveydetails/widget/answer_text_field_widget.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    TextField(
+      onChanged: (text) => _onTextChanged(-1, text),
+      controller: TextEditingController(),
+    );
```
Comment (manh-t) [inline lib/screens/surveydetails/widget/answer_text_field_widget.dart]: @ryan-conway not sure why you create a widget here without any usage?

**Review by manh-t — COMMENTED** [review]
just one concern

File: lib/screens/surveydetails/widget/answer_text_field_widget.dart (hunk 2 above)
Comment (ryan-conway) [reply lib/screens/surveydetails/widget/answer_text_field_widget.dart]: My mistake, I was checking how `TextField` works and forgot to remove it 🙈 Updated in [4b0cdcf](https://github.com/nimblehq/flutter-ic-lydia-ryan/pull/89/commits/4b0cdcf90b21f67e9c862b40df27636b5fe94f60)

**Review by manh-t — APPROVED** [review]

**Pull request merged**